src/c4v/c4v_cli.py

```diff
@@ -11,91 +11,123 @@
+    if not os.path.isdir(DEFAULT_FILES_FOLDER):
+        click.echo(
+            f"[INFO] Creating local files folder at: {DEFAULT_FILES_FOLDER}", err=True
+        )
+        try:
+            os.mkdir(DEFAULT_FILES_FOLDER)
+        except Exception as e:
+            print(e)
+
 
 @c4v_cli.command()
-@click.option('--files',    is_flag=True,   help="Interpret url list as files instead of urls, so urls are retrieved from such files. The file is expected to be formatted such that there's one url per line")
-@click.option('--output',   default=None,   help="Output file to store results")
-@click.option('--max-len',  default=-1,     help="Truncate scraped content's body to a maximum lenght if provided.")
-@click.option('--pretty',   is_flag=True,   help="Print data formatted as human readable")
-@click.argument('urls', nargs = -1)
-def scrape(urls : List[str] = None, files : bool = None, output : str = None, max_len : int = -1, pretty : bool = False):
+@click.option(
+    "--files",
+    is_flag=True,
+    help="Interpret url list as files instead of urls, so urls are retrieved from such files. The file is expected to be formatted such that there's one url per line",
+)
+@click.option(
+    "--loud", is_flag=True, help="Print scraped data to stdio as it is being scraped"
+)
+@click.option(
+    "--up-to",
+    default=-1,
+    help="Maximum amount of elements to scrape. Scrape all if no argument given",
+)
+@click.argument("urls", nargs=-1)
+def scrape(
+    urls: List[str] = None, files: bool = None, loud: bool = False, up_to: int = -1
+):
     """
-        Use this command to run a scraping process.
-        Parameters:
-            + urls : [str] = List of urls or files to get urls from
-            + files : bool = if urls are in files or not, defaults to false
-            + output : str = where to store output 
-            + max_len : int = max body size to print when pretty printing. Defaults to max size possible when not provided
-            + pretty : bool = If print should be human readable
+        Use this command to run a scraping process. Scraped urls are retrieved from database by default.\n
+        If provided with a list of urls, then those urls are the ones to be scraped and stored. \n
+        You can pass the --files flag to interpret those paths as paths to files storing the actual urls,\n
+        formated as one url per line.\n
+        Parameters:\n
+            + urls : [str] = List of urls or files to get urls from\n
+            + files : bool = if urls are in files or not, defaults to false\n
+            + up_to : int = maximum amount of urls to scrape in this pass. Scrape all non-scraped if not provided\n
+            + loud : bool = if should print scraped data once a scraping is finished\n
     """
 
-    # Check for errors
-    #   check if urls were provided
-    if not urls: 
-        click.echo("No urls to scrape were provided")
-        return
+    db_manager = SqliteManager(DEFAULT_DB)
```

Comment:
I wonder if we should have all the logic in these methods in a high level scrap manager class that represents some sort of Main for scrapping.

That would allow us to have 2 things:
- a thinner cli logic
- a reusable logic that clients of the library could use programmatically without calling methods intended for the CLI

Reply:
Yes, in fact I'm working on it on another branch for another task, so it's coming :)